hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -126,9 +171,205 @@ public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckExcept
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
+        requireNonNull(context);
         final var op = context.body().tokenRejectOrThrow();
+        final var rejectingAccountID = op.ownerOrElse(context.payer());
         final var rejections = op.rejections();
-        // Todo: Implement the logic for token rejection
+
+        final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
+        final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
+        final var tokensConfig = context.configuration().getConfigData(TokensConfig.class);
+        validateTrue(tokensConfig.tokenRejectEnabled(), NOT_SUPPORTED);
+        validateTrue(rejections.size() <= ledgerConfig.tokenRejectsMaxLen(), INVALID_TRANSACTION_BODY);
+
+        final var accountStore = context.writableStore(WritableAccountStore.class);
+        final var rejectingAccount = getIfUsableForAliasedId(
+                rejectingAccountID, accountStore, context.expiryValidator(), INVALID_ACCOUNT_ID);
+
+        // Process the rejections to prepare for the transfers and the removal of allowances
+        final var processedRejections =
+                processRejectionsForTransferAndAllowancesRemoval(rejections, context, rejectingAccount);
+        final var body = CryptoTransferTransactionBody.newBuilder()
```

Comment:
Fully agreed. Thanks for the suggestion, its a great one. Since we only use the tokenTransferLists in those steps, no reason to have the full body there. I've refactored it.